packages/token-bridge-sdk/src/deposits/fetchDepositsFromSubgraph.ts

```diff
@@ -0,0 +1,121 @@
+import { gql } from '@apollo/client'
+import { getL1SubgraphClient } from '../util/subgraph'
+
+export type FetchDepositsFromSubgraphResult = {
+  receiver: string
+  sender: string
+  sequenceNumber: string
+  timestamp: string
+  transactionHash: string
+  type: 'EthDeposit' | 'TokenDeposit'
+  isClassic: boolean
+  id: string
+  ethValue: string
+  tokenAmount?: string
+  blockCreatedAt: string
+  l1Token?: {
+    symbol: string
+    decimals: string
+    id: string
+    name: string
+    registeredAtBlock: string
+  }
+}
+
+export const fetchDepositsFromSubgraph = async ({
+  address,
+  fromBlock,
+  toBlock,
+  l2ChainId,
+  pageSize = 10,
+  pageNumber = 0,
+  searchString = ''
+}: {
+  address: string
+  fromBlock?: number
+  toBlock?: number
+  l2ChainId: number
+  pageSize?: number
+  pageNumber?: number
+  searchString?: string
+}): Promise<FetchDepositsFromSubgraphResult[]> => {
+  if (fromBlock === 0 && toBlock === 0) {
+    return []
+  }
+
+  const res = await getL1SubgraphClient(l2ChainId).query({
+    query: gql`{
+        deposits(
+          where: {
+            sender: "${address}",
+            blockCreatedAt_gte: ${fromBlock},
+            blockCreatedAt_lte: ${toBlock}
+            ${searchString ? `transactionHash_contains: "${searchString}"` : ''}
+          }
+          orderBy: blockCreatedAt
+          orderDirection: desc
+          first: ${pageSize},
+          skip: ${pageNumber * pageSize}
+        ) {
+          receiver
+          sender
+          sequenceNumber
+          timestamp
+          tokenAmount
+          transactionHash
+          type
+          isClassic
+          id
+          ethValue
+          blockCreatedAt
+          l1Token {
+            symbol
+            decimals    
+            id
+            name
+            registeredAtBlock
+          }                  
+        }
+      }
+    `
+  })
+
+  const transactions: FetchDepositsFromSubgraphResult[] = res.data.deposits.map(
+    (tx: FetchDepositsFromSubgraphResult) => {
+      const {
```

Comment:
You're getting all properties here, we can have have it shorter:
```
const {
   { l1Token: { id, symbol },
   ...rest,
} = tx

return {
  ...rest,
  l1Token: {
    id,
    symbol,
  }
}
```

Reply:
`l1Token` was optional here so couldn't extract it in a restructured way, but I've updated the redundant code to be very compact now.